lib/index.js

```diff
@@ -182,7 +190,14 @@ internals.Client.prototype._request = function (method, url, options, relay, _tr
     }
 
     if (this._emit) {
-        this._emit('request', uri, options);
+        const payload = uri;
```

Comment:
Will this need to change if we update `applyUrlToOptions` to be like node core as well as to copy over the username/password?

Reply:
I think that depends on how much we are willing to change the payload on this event. Once I add the auth prop to the uri, we could replace `username`/`password` with the combined property like the WHATWG parser does. 

I'll leave that decision to you, since it will determine the semver level of this change.

Reply:
I'd prefer to pass whatever the `uri` is set to at this stage into the event. To avoid breakage we could still copy the `username`/`password`, `searchParams`, and `host` in `applyUrlToOptions`. If that will cause issues to copy over then maybe we shouldn't copy them and release this as a semver major.

Reply:
Okay, I modified `applyUrlToOptions` to pull in username/password, origin, and searchParams. I did not pull in `host` because of the issues with port numbers when both host/hostname are present. 

Other than `host`/`hostname` I believe the contract on the emit call is intact.

Reply:
Great, thank you